voltdb/src/main/java/com/yahoo/ycsb/db/voltdb/ConnectionHelper.java

```diff
@@ -0,0 +1,215 @@
+    ClientConnection conn = clientMapping.get(clientId);
+    if (conn != null) {
+      return conn.mclient;
+    }
+    if (activeConnection != null && activeConnection.mconnectionCount.get() <= THREADS_PER_CLIENT) {
+      activeConnection.connect();
+      clientMapping.put(clientId, activeConnection);
+      return activeConnection.mclient;
+    }
+    ClientConfig config = new ClientConfig(user, password);
+    config.setMaxTransactionsPerSecond(ratelimit);
+    Client client = ClientFactory.createClient(config);
+    connect(client, servers);
+    activeConnection = new ClientConnection(client);
+    clientMapping.put(clientId, activeConnection);
+    return client;
+  }
+
+  public static void disconnect(Long clientId) {
+    ClientConnection connection = clientMapping.get(clientId);
+    if (connection != null) {
+      connection.disconnect();
+    }
+  }
+
+  /**
+   * Connect to a single server with retry. Limited exponential backoff. No
+   * timeout. This will run until the process is killed if it's not able to
+   * connect.
+   *
+   * @param server hostname:port or just hostname (hostname can be ip).
+   */
+  private static void connectToOneServerWithRetry(final Client client, String server) {
+    int sleep = 1000;
+    while (true) {
+      try {
+        client.createConnection(server);
+        break;
+      } catch (Exception e) {
+        System.err.printf("Connection failed - retrying in %d second(s).\n", sleep / 1000);
+        try {
+          Thread.sleep(sleep);
+        } catch (java.lang.InterruptedException e2) {
+          e.printStackTrace();
+        }
+        if (sleep < 8000) {
+          sleep += sleep;
+        }
+      }
+    }
+    System.out.printf("Connected to VoltDB node at: %s.\n", server);
+  }
+
+  /**
+   * See if DB servers are present on the network.
+   * 
+   * @return true or false
+   */
+  public static boolean checkDBServers(String servernames) {
+
+    String[] serverNamesArray = servernames.split(",");
+
+    boolean dbThere = false;
+
+    Socket socket = null;
+    try {
+      // Connect
+      socket = new Socket(serverNamesArray[0], VOLTDB_DEFAULT_PORT);
+      dbThere = true;
+    } catch (IOException connectFailed) {
+      dbThere = false;
+    } finally {
+      if (socket != null) {
+        try {
+          socket.close();
+        } catch (IOException ignore) {
+          // Ignore.
+        }
+      }
+      socket = null;
+    }
+
+    return dbThere;
+
+  }
+
+  /**
+   * Connect to a set of servers in parallel. Each will retry until connection.
+   * This call will block until all have connected.
+   *
+   * @param servers A comma separated list of servers using the hostname:port
+   *                syntax (where :port is optional).
+   * @throws InterruptedException if anything bad happens with the threads.
+   */
+  private static void connect(final Client client, String servers) throws InterruptedException {
+    System.out.println("Connecting to VoltDB...");
```

Comment:
Could you use a logging framework instead of System.out/System.err? slf4j if you don't have another preference.

Reply:
Fixed in 0ebc1ec